installer/generate_secrets.py

```diff
@@ -1,38 +1,50 @@
 #!/usr/bin/env python3
 from base64 import b64encode
 import bcrypt
+from collections import defaultdict
 from cryptography.fernet import Fernet
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization
+from datetime import datetime, timezone
 import json
 import os
+from pathlib import Path
 import secrets
 import yaml
 
 
-def input_field(component, name, description):
-    prompt_string = f"[{component} {name}] ({description}): "
-    return input(prompt_string)
+def input_field(s, component, name, description):
+    default = s[component].get(name, "")
+    prompt_string = f"[{component} {name}] ({description}): [current: {default}] "
+    input_string = input(prompt_string)
 
+    if input_string:
+        s[component][name] = input_string
 
-def generate_tap_secrets():
-    fname = input_field(
-        "TAP",
-        "google_creds.json",
-        "file containing google service account credentials",
-    )
 
-    with open(fname, "r") as f:
-        return {
-            "slack_webhook_url": input_field(
-                "TAP", "slack_webhook_url", "slack webhook url for querymonkey"
-            ),
-            "google_creds.json": b64encode(f.read().encode()).decode(),
-        }
+def input_file(s, component, name, description):
+    current = s.get(component, {}).get(name, "")
+    print(f"[{component} {name}] ({description})")
+    print(f"Current contents:\n{current}")
+    prompt_string = f"New filename with contents (empty to not change): "
+    fname = input(prompt_string)
+
+    if fname:
+        with open(fname, "r") as f:
+            s[component][name] = f.read()
+
 
+def set_generated_secret(s, component, name, new_value):
+    regenerate = True
+    if regenerate:
```

Comment:
Do you want to pass regenerate as an argument to the method?  If not, why bother with the if?

Reply:
Ah yes, I forgot about that, I want to pass that in as an argument to the whole program, will do